### ISSUE TITLE
Future compatibility warning ("CloneToAny" cannot be made into an object)

### ISSUE THREAD

**Issue by jonas-schievink** [chris-morgan/anymap#31]
Compiling anymap on the current nightly results in this warning: https://github.com/rust-lang/rust/issues/51443

```
warning: the trait `any::CloneToAny` cannot be made into an object                                                                                                                
  --> src/any.rs:15:5                                                                                                                                                             
   |                                                                                                                                                                              
15 |     fn clone_to_any_send(&self) -> Box<CloneAny + Send> where Self: Send;                                                                                                    
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                    
   |                                                                                                                                                                              
   = note: #[warn(where_clauses_object_safety)] on by default                                                                                                                     
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!                                              
   = note: for more information, see issue #51443 <https://github.com/rust-lang/rust/issues/51443>                                                                                
   = note: method `clone_to_any_send` references the `Self` type in where clauses                                                                                                 
```
(same warning for all `clone_to_any_X` methods)

**Comment by jonas-schievink** [comment]
Yeah that seems fine, thanks! FWIW the typemap crate is also affected and has a very unsafe fix here: https://github.com/reem/rust-typemap/pull/41

**Comment by chris-morgan** [comment]
Well, although it’s acknowledged to be a spurious warning, nothing’s been done about it, so I’ve finally caved and implemented it a different way, adding some more `unsafe`, but avoiding this troublesome future-compatibility warning.

Fixed in 521fbfe.
